source/extensions/filters/network/zookeeper_proxy/zookeeper_filter.cc

```diff
@@ -0,0 +1,110 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_filter.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+#include "common/common/logger.h"
+
+#include "extensions/filters/network/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+ZooKeeperFilterConfig::ZooKeeperFilterConfig(const std::string& stat_prefix, Stats::Scope& scope)
+    : scope_(scope), stat_prefix_(stat_prefix), stats_(generateStats(stat_prefix, scope)) {}
+
+ZooKeeperFilter::ZooKeeperFilter(ZooKeeperFilterConfigSharedPtr config)
+    : config_(std::move(config)) {}
+
+void ZooKeeperFilter::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) {
+  read_callbacks_ = &callbacks;
+}
+
+Network::FilterStatus ZooKeeperFilter::onData(Buffer::Instance& data, bool) {
+  doDecode(data);
+  return Network::FilterStatus::Continue;
+}
+
+Network::FilterStatus ZooKeeperFilter::onWrite(Buffer::Instance& data, bool) {
+  doDecode(data);
+  return Network::FilterStatus::Continue;
+}
+
+Network::FilterStatus ZooKeeperFilter::onNewConnection() { return Network::FilterStatus::Continue; }
+
+void ZooKeeperFilter::doDecode(Buffer::Instance& buffer) {
+  // Clear dynamic metadata.
```

Comment:
dont you need a fallback in case decoding fails?

Reply:
@rshriram that's handled in the decoder itself... There's a test case for failing to decode.. Or what do you mean?